cmd/controller/controller.go

```diff
@@ -99,11 +106,212 @@ func NewController(store *store.AuthStore) *Controller {
+	// 		key := fmt.Sprintf("%s;%s;%s", origin, target, ref.For)
+
+	// 		if _, ok := clusterRefConsumers[key]; !ok {
+	// 			clusterRefConsumers[key] = map[string][]v1a1.Subject{
+	// 				"": {},
+	// 			}
+	// 		}
+	// 		if len(crc.ClassNames) > 0 {
+	// 			c.log.Info("Opt in to ClusterReferenceConsumer ClassNames", "ClusterReferenceConsumer", crc.Name)
+	// 			for _, className := range crc.ClassNames {
+	// 				if _, ok := clusterRefConsumers[key][className]; !ok {
+	// 					clusterRefConsumers[key][className] = []v1a1.Subject{}
+	// 				}
+	// 				clusterRefConsumers[key][className] = append(clusterRefConsumers[key][className], crc.Subject)
+	// 			}
+	// 		} else {
+	// 			clusterRefConsumers[key][""] = append(clusterRefConsumers[key][""], crc.Subject)
+	// 		}
+	// 	}
+	// }
+
 	return ctrl.Result{}, nil
 }
 
-func generateQueueKey(fromGroup, fromResource, toGroup, toResource, forReason string) types.NamespacedName {
-	nn := types.NamespacedName{Name: forReason}
-	nn.Namespace = fmt.Sprintf("%s/%s-%s/%s")
-	return nn
+// func generateQueueKey(fromGroup, fromResource, toGroup, toResource, forReason string) types.NamespacedName {
+// 	nn := types.NamespacedName{Name: forReason}
+// 	nn.Namespace = fmt.Sprintf("%s/%s-%s/%s")
+// 	return nn
+// }
+
+type reference struct {
+	Group         string
+	Resource      string
+	FromNamespace string
+	ToNamespace   string
+	Name          string
+}
+
+func (c *Controller) getReferences(list *unstructured.UnstructuredList, path string) ([]reference, error) {
+	refs := []reference{}
+	for _, item := range list.Items {
+		j := jsonpath.New("test")
+		err := j.Parse(fmt.Sprintf("{%s}", path))
+		if err != nil {
+			c.log.Error(err, "error parsing JSON Path")
+			return refs, err
+		}
+		results := new(bytes.Buffer)
+		err = j.Execute(results, item.UnstructuredContent())
+		if err != nil {
+			c.log.Error(err, "error finding results with JSON Path")
+			return refs, err
+		}
+
+		rawRefs := strings.Split(results.String(), " ")
+
+		for _, rr := range rawRefs {
+			jr := map[string]string{}
+			err = json.Unmarshal([]byte(rr), &jr)
+			if err != nil {
+				c.log.Error(err, "error finding results with JSON Path")
+			}
+			group, hasGroup := jr["group"]
+			if !hasGroup {
+				c.log.Info("Missing group in reference", "ref", jr)
+				continue
+			}
+			resource, hasResource := jr["resource"]
+			if !hasResource {
+				kind, hasKind := jr["kind"]
+				if !hasKind {
+					c.log.Info("Missing kind or resource in reference", "ref", jr)
+					continue
+				}
+				gvr, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Group: group, Version: "v1", Kind: kind})
+				resource = gvr.Resource
+			}
+
+			namespace, hasNamespace := jr["namespace"]
+			if !hasNamespace {
+				namespace = item.GetNamespace()
+			}
+
+			name, hasName := jr["name"]
+			if !hasName {
+				c.log.Info("Missing name in reference", "ref", jr)
+				continue
+			}
+			// TODO: check if the referenced resource exist ?
```

Comment:
I don't think there's any need for that, k8s authorization has always been ambivalent to whether or not something actually exists, this is an eventually consistent system and it's likely that the resource may not exist yet when we'd be doing this check. Would not have to start watching yet another kind of resource here.

Reply:
sgtm, thanks